dm/pkg/binlog/event/event.go

```diff
@@ -736,21 +736,23 @@ func GenMariaDBGTIDListEvent(header *replication.EventHeader, latestPos uint32,
 		return nil, terror.ErrBinlogWriteBinaryData.AnnotateDelegate(err, "write Number of GTIDs %d", numOfGTIDs)
 	}
 
-	for _, mGTID := range mariaDBGSet.Sets {
-		// Replication Domain ID, 4 bytes
-		err = binary.Write(payload, binary.LittleEndian, mGTID.DomainID)
-		if err != nil {
-			return nil, terror.ErrBinlogWriteBinaryData.AnnotateDelegate(err, "write Replication Domain ID %d", mGTID.DomainID)
-		}
-		// Server_ID, 4 bytes
-		err = binary.Write(payload, binary.LittleEndian, mGTID.ServerID)
-		if err != nil {
-			return nil, terror.ErrBinlogWriteBinaryData.AnnotateDelegate(err, "write Server_ID %d", mGTID.ServerID)
-		}
-		// GTID sequence, 8 bytes
-		err = binary.Write(payload, binary.LittleEndian, mGTID.SequenceNumber)
-		if err != nil {
-			return nil, terror.ErrBinlogWriteBinaryData.AnnotateDelegate(err, "write GTID sequence %d", mGTID.SequenceNumber)
+	for _, set := range mariaDBGSet.Sets {
```

Comment:
in line 733 there's `numOfGTIDs` which should also be updated. Before fix it, find out why the test didn't reveal the problem.

Reply:
<img width="572" alt="image" src="https://github.com/pingcap/tiflow/assets/23556834/a9f4caf4-6415-4bbd-aaf9-1f7dbf50e527">
This function is used in test only 😅

Reply:
Let me take a closer look at unit test

Reply:
I found the reason why the unit test succeeds. All executions of GenMariaDBGTIDListEvent will execute
https://github.com/pingcap/tiflow/blob/6453fef89e71d3a76dd0b47f784c1e370dfb37e0/dm/pkg/binlog/event/util.go#L162 

This function will decode the event into the "correct" form to compare with other results. There is a high probability that other results will also be decoded through it.

I will add an unit test case for it.

Reply:
https://github.com/pingcap/tiflow/pull/10753/commits/3b094a0358b60403809b17c1c814d50f29887819